### PR TITLE
Add post-release validation workflow

### PR DESCRIPTION
**Pull request by ni-jfitzger**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [ ] ~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~
- [ ] ~I've added tests applicable for this pull request~

### What does this Pull Request accomplish?
Add a workflow to test installing the latest released nimi-python modules from PyPI and running examples.
It will trigger when a [release is published on the repo](https://github.com/ni/nimi-python/releases).

Notes:
* To keep things simple, I'm not using tox-system_tests.ini and I'm only running on one version of Python.
* I'm using the built-in venv to provide a clean environment for every run. virtualenv is superior, but for a single Python version, venv should be good enough.
* [Only users with write access should be able to manually trigger the workflow](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow#configuring-a-workflow-to-run-manually)
------
#### Why?

Our internal release template mentions installing modules from PyPI and running examples after publishing a release, to validate it. This also requires installing every driver. The whole thing is tedious; we should just have a workflow.

### List issues fixed by this Pull Request below, if any.

None

### What testing has been done?
Implemented a [temporary tweak](https://github.com/ni/nimi-python/pull/2058/files/f8d766dd43aabcb8c69910e4902e57280e90d054..4b7edb99679190954f1da68a9865235c77242728) to trigger the workflow from this PR for the 1.4.8 release.
[All tests passed](https://github.com/ni/nimi-python/actions/runs/8862289563).

Running off a manual trigger or release event remains untested.

### CODE REVIEW

**Comment by ni-jfitzger** [comment]
This isn't ready for merge yet, because our post-release change needs to go in first.

File: .github/actions/run_examples_using_pypi_uploads/action.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Run examples using pypi uploads
```
Comment (marcoskirsch) [inline .github/actions/run_examples_using_pypi_uploads/action.yml:1]:
```suggestion
name: Run examples using PyPI uploads
```

File: .github/actions/run_examples_using_pypi_uploads/action.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+description: Install module(s) from PyPI and run examples. Intended for release validation.
+inputs:
+  module_name:
+    description: module name
```
Comment (marcoskirsch) [inline .github/actions/run_examples_using_pypi_uploads/action.yml:5]:
```suggestion:
    description: module name (e.g. "nidcpower")
```

File: .github/actions/run_examples_using_pypi_uploads/action.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    required: true
+    default: ""
+  module_version:
+    description: module version
```
Comment (marcoskirsch) [inline .github/actions/run_examples_using_pypi_uploads/action.yml:9]:
```suggestion
    description: module version (e.g. "1.8.1")
```

File: .github/actions/run_examples_using_pypi_uploads/action.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    required: true
+    default: ""
+
+runs:
+  using: composite
+  steps:
+    # The local wheel installation and single environment would have made using tox-system_tests.ini difficult.
+    # To keep things simple, we just use pip and pytest directly for a single Python version.
+    - run: py -3.12 -m venv --clear .venv
```
Comment (marcoskirsch) [inline .github/actions/run_examples_using_pypi_uploads/action.yml:18]: can you think of any way to make it less likely that this version becomes stale as new Python releases come out?
any way to single source?

(if it's too complicated, ok to leave alone)

File: .github/workflows/check_latest_release.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+on:
+  release:
+    types: [released]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    inputs:
+      release_tag:
+        description: The GitHub release tag corresponding to the pypi releases to test
```
Comment (marcoskirsch) [inline .github/workflows/check_latest_release.yml:11]: let's capitalize PyPI (and other things) correctly in places were it's ok to mix caps.

File: .github/actions/run_examples_using_pypi_uploads/action.yml (hunk 4 above)
Comment (marcoskirsch) [reply .github/actions/run_examples_using_pypi_uploads/action.yml:18]: Maybe it would help if Python version was an input as well?

**Review by marcoskirsch — CHANGES_REQUESTED** [review]

File: .github/actions/run_examples_using_pypi_uploads/action.yml (hunk 4 above)
Comment (ni-jfitzger) [reply .github/actions/run_examples_using_pypi_uploads/action.yml:18]: Well, when I add a new version, I generally do a search for the last version, (3.12, 312), so I expect it won't be a problem. Version support changes are currently completely manual. It's one more thing we could stand to have a tool for.

File: .github/actions/run_examples_using_pypi_uploads/action.yml (hunk 2 above)
Comment (ni-jfitzger) [reply .github/actions/run_examples_using_pypi_uploads/action.yml:5]:
```suggestion
    description: module name (e.g. "nidcpower")
```

File: .github/workflows/check_latest_release.yml (hunk 5 above)
Comment (ni-jfitzger) [reply .github/workflows/check_latest_release.yml:11]:
```suggestion
        description: The GitHub release tag corresponding to the PyPI releases to test
```

File: .github/workflows/check_latest_release.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+          - nidmm
+          - niscope
+          - nimodinst
+          - nise
+          - niswitch
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v3
+      # NOTE: we don't upload test coverage for this
+      - name: run examples using pypi uploads
```
Comment (ni-jfitzger) [inline .github/workflows/check_latest_release.yml:44]:
```suggestion
- name: run examples using PyPI uploads
```

File: .github/actions/run_examples_using_pypi_uploads/action.yml (hunk 4 above)
Comment (ni-jfitzger) [reply .github/actions/run_examples_using_pypi_uploads/action.yml:18]: I created #2060 to help address your concerns.

**Review by marcoskirsch — APPROVED** [review]

**Pull request merged**